mobilenet-burn/src/model/inverted_residual.rs

```diff
@@ -0,0 +1,94 @@
+use super::conv_norm::NormalizationLayer;
+use super::conv_norm::NormalizationType;
+use super::conv_norm::{Conv2dNormActivation, Conv2dNormActivationConfig};
+use alloc::vec;
+use alloc::vec::Vec;
+use burn::config::Config;
+use burn::nn::conv::Conv2dConfig;
+use burn::nn::BatchNormConfig;
+use burn::tensor::Tensor;
+use burn::{module::Module, nn::conv::Conv2d, tensor::backend::Backend};
+#[derive(Module, Debug)]
+enum InvertedResidualSequentialType<B: Backend> {
+    Conv(Conv2d<B>),
+    ConvNormActivation(Conv2dNormActivation<B>),
+    NormLayer(NormalizationLayer<B, 4>),
+}
```

Comment:
I know that the pytorch implementation uses `nn.Sequential`, but the number of layers isn't even variable. The only thing that changes is an additional `Conv2dNormActivation` when expand ratio != 1. I think sequential is overused in general 😄 

In this case I would probably remove the enum and layers vector to have a fixed definition. For example:

```rust
#[derive(Module, Debug)]
pub struct PointWiseLinear<B: Backend> {
   conv: Conv2d<B>,
   norm: NormalizationLayer<B, 4>,
}

#[derive(Module, Debug)]
pub struct InvertedResidual<B: Backend> {
    use_res_connect: bool,
    pw: Option<Conv2dNormActivation<B>>, // pointwise, only when expand ratio != 1
    dw: Conv2dNormActivation<B>,
    pw_linear: PointWiseLinear<B>,
}
```

And for the weights we'll be able to remap them anyway.

Reply:
Done